hashicorp/dev-portal
# chore: trim product-tutorials static props

## 🔗 Relevant links

- [Preview link](https://dev-portal-git-BRANCH_NAME-hashicorp.vercel.app/PATH_TO_VIEW) 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Trims `getStaticProps` data size on "product tutorial landing" views, by formatting Tutorial and Collection data for components before returning from `getStaticProps`. Previously, raw Tutorial and Collection data was returned and then formatted on the client.

## 🤷 Why

To reduce page load times, by reducing the amount of data transferred and reducing the amount of client-side data processing. May also be a slight improvement for the developer experience of working on the related views and components.

## 🧪 Testing

- [ ] Visit pages listed in the table below
    - Open browser dev tools, filter for network requests for `json` files
    - Confirm the resource size of each `tutorials.json?product=<productSlug>` file is ~ as listed in the table below
    - Visit the same pages on `developer.hashicorp.com` to confirm the previous resource sizes

| View                              | ~ kB Before | ~ kB After | Shows Sitemap |
| :-------------------------------- | ----------: | ---------: | ------------- |
| [/hcp/tutorials][hcp]             |        76.3 |        9.8 | No            |
| [/terraform/tutorials][terraform] |       167.9 |       61.4 | Yes           |
| [/packer/tutorials][packer]       |        27.9 |       10.4 | Yes           |
| [/consul/tutorials][consul]       |        64.5 |       44.2 | Yes           |
| [/vault/tutorials][vault]         |       139.4 |       51.9 | Yes           |
| [/boundary/tutorials][boundary]   |        46.7 |        6.6 | No            |
| [/nomad/tutorials][nomad]         |        62.1 |       27.6 | Yes           |
| [/waypoint/tutorials][waypoint]   |        25.0 |       11.1 | Yes           |
| [/vagrant/tutorials][vagrant]     |        18.2 |        6.1 | No            |

> **Note**: Sitemap data is already optimized (thanks to [#420](https://github.com/hashicorp/dev-portal/pull/420)). But even when optimized, some of our sitemaps are large enough that they seem to still contribute significant weight to static props.

## 💭 Anything else?

Not at the moment.

[task]: https://app.asana.com/0/1202097197789424/1202225913259472/f
[preview]: https://dev-portal-git-zstrim-product-tutorials-gsp-weight-hashicorp.vercel.app/
[hcp]: https://dev-portal-git-zstrim-product-tutorials-gsp-weight-hashicorp.vercel.app/hcp/tutorials
[terraform]: https://dev-portal-git-zstrim-product-tutorials-gsp-weight-hashicorp.vercel.app/terraform/tutorials
[packer]: https://dev-portal-git-zstrim-product-tutorials-gsp-weight-hashicorp.vercel.app/packer/tutorials
[consul]: https://dev-portal-git-zstrim-product-tutorials-gsp-weight-hashicorp.vercel.app/consul/tutorials
[vault]: https://dev-portal-git-zstrim-product-tutorials-gsp-weight-hashicorp.vercel.app/vault/tutorials
[boundary]: https://dev-portal-git-zstrim-product-tutorials-gsp-weight-hashicorp.vercel.app/boundary/tutorials
[nomad]: https://dev-portal-git-zstrim-product-tutorials-gsp-weight-hashicorp.vercel.app/nomad/tutorials
[waypoint]: https://dev-portal-git-zstrim-product-tutorials-gsp-weight-hashicorp.vercel.app/waypoint/tutorials
[vagrant]: https://dev-portal-git-zstrim-product-tutorials-gsp-weight-hashicorp.vercel.app/vagrant/tutorials
